src/Configuration/Properties/Property.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelCom\Integration\Configuration\Properties;
+
+class Property
+{
+    public function __construct(
+        private readonly string $name,
+        private readonly PropertyType $type,
+        private readonly ?string $description = null,
+        private readonly array $enum = [],
+        private readonly array $meta = [],
+    ) {
+    }
+
+    public function toArray(): array
+    {
+        return [
+            $this->name => array_filter([
+                'type' => $this->type->value,
+                'description' => $this->description,
+                'enum' => $this->enum,
+                'meta' => $this->meta,
```

Comment:
I'm thinking... we already have a structure for what's expected inside the meta, right? There is `password: true`, so is `hint: "some hint"` (I might be mistaken if it was `hint`, please check in the api code).

Shall we suspend the open $this->meta field in favor of more refined bunch, such as `isPassword` and `hint` (or whatever it is called)?

### Even more obvious property types
In terms of the forms we construct, we currently have a `Text` field, a `Password` field and a `Select` field, right? We can make the usage of the configuration properties even easier if we define these as separate classes based on the same `Property` or `Field` classes. That will help us to expand the catalogue with more. For example, if we come up with some indication in the `meta` for multi-select choice, or date/time picker, we could define new classes with ease, and they will be very easy to use.

Basically my proposal is to hide the json-schema as much as possible from the developer-user. They only have to know they are constructing a configuration form, the format used for the transport ain't important for that task. 

So logically, a _Text_ field could accept a name, description, hint (or whatever it is called). A _Password_ field could extend the Text, but modify the serialization of the field into json-schema by setting meta.password: true.